zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/migration/MigrateSequentialMultiInstanceBodyTest.java

```diff
@@ -1034,4 +1034,126 @@ public void shouldRaiseIncidentOnCompletionAfterChangingOutputCollection() {
+                            t.zeebeJobType("B")
+                                .multiInstance(
+                                    b ->
+                                        b.sequential()
+                                            .zeebeInputCollectionExpression("input2")
+                                            .zeebeInputElement("index")
+                                            .zeebeOutputElementExpression("index")
+                                            .zeebeOutputCollection("results")))
+                    .endEvent("multi_instance_target_process_end")
+                    .done())
+            .deploy();
+
+    final long targetProcessDefinitionKey =
+        extractProcessDefinitionKeyByProcessId(deployment, targetProcessId);
+
+    final var processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(sourceProcessId)
+            .withVariable("input1", Arrays.asList("1", "2", "3"))
+            .create();
+
+    final var resultsVariable =
+        RecordingExporter.variableRecords(VariableIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .withName("input1")
+            .withValue("[\"1\",\"2\",\"3\"]")
+            .getFirst();
+
+    final var job =
+        RecordingExporter.jobRecords(JobIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .withType("A")
+            .getFirst();
+
+    ENGINE.job().withKey(job.getKey()).complete();
+
+    final var secondJob =
+        RecordingExporter.jobRecords(JobIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .withType("A")
+            .withElementId("serviceTask1")
+            .skip(1)
+            .getFirst();
+
+    // when
+    ENGINE
+        .processInstance()
+        .withInstanceKey(processInstanceKey)
+        .migration()
+        .withTargetProcessDefinitionKey(targetProcessDefinitionKey)
+        .addMappingInstruction("serviceTask1", "serviceTask2")
+        .migrate();
+
+    // then
+    // after migrating, we can complete the second job
+    ENGINE.job().ofInstance(processInstanceKey).withKey(secondJob.getKey()).complete();
+
+    // but that raises an incident
+    final var incident =
+        RecordingExporter.incidentRecords(IncidentIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst();
+    Assertions.assertThat(incident.getValue())
+        .hasErrorMessage(
+            "Expected result of the expression 'input2' to be 'ARRAY', but was 'NULL'. "
+                + "The evaluation reported the following warnings:\n"
+                + "[NO_VARIABLE_FOUND] No variable found with name 'input2'");
+
+    // we can resolve the incident by creating the new input collection variable 'input2'
+    ENGINE
+        .variables()
+        .ofScope(resultsVariable.getValue().getScopeKey())
+        .withDocument("{\"input2\": [1, null, null]}")
```

Comment:
🔧 This input is a bit strange. Let's change it to:
```suggestion
        .withDocument("{\"input2\": [1, 2, 3]}")
```